packages/expo-modules-core/android/src/main/java/expo/modules/kotlin/types/TypeConverterProvider.kt

```diff
@@ -91,6 +91,16 @@ object TypeConverterProviderImpl : TypeConverterProvider {
       return EnumTypeConverter(kClass as KClass<Enum<*>>, type.isMarkedNullable)
     }
 
+    if (kClass.isSubclassOf(Either::class)) {
```

Comment:
Maybe move it down after `kClass.isSubclassOf(Record::class)` check as records will be used more often

Reply:
I'll move it, but just for the record it really doesn't matter - that code is run once during the initialization.